### PR TITLE
assumptions: add refine handlers for sinh, cosh, and tanh

### PR DESCRIPTION
**Pull request by HITESH-S-P**
Add new refine handlers for hyperbolic functions (sinh, cosh, tanh) that simplify expressions under assumptions:
- sinh(x) -> 0 when x is zero
- cosh(x) -> 1 when x is zero
- tanh(x) -> 0 when x is zero
- sinh(I*x) -> I*sin(x) when x is real
- cosh(I*x) -> cos(x) when x is real

#### References to other Issues or PRs

Part of #27888

#### Brief description of what is fixed or changed
Added refine handlers for hyperbolic functions (`sinh`, `cosh`, `tanh`) so that
`refine()` can simplify expressions involving these functions under assumptions.
**New simplifications enabled:**
- `refine(sinh(x), Q.zero(x))` → `0`
- `refine(cosh(x), Q.zero(x))` → `1`
- `refine(tanh(x), Q.zero(x))` → `0`
- `refine(sinh(I*x), Q.real(x))` → `I*sin(x)`
- `refine(cosh(I*x), Q.real(x))` → `cos(x)`
This follows the same pattern as existing handlers for `sin`/`cos` (PR #28873)
and `tan`/`cot` (PR #29324).

#### Other comments
I observed hyperbolic functions were
unhandled by `refine` after checking all open and closed PRs tagged under
`assumptions.refine`.

#### AI Generation Disclosure

- **Tool used:** Antigravity
- **AI-generated parts:** Refined the code formatting and wording in `sympy/assumptions/refine.py`.
- **Human-authored/reviewed parts:** I designed the underlying math and simplification logic, wrote the unit tests in `sympy/assumptions/tests/test_refine.py` and verified all tests pass locally.

#### Release Notes

* assumptions
  * Added refine handlers for `sinh`, `cosh`, and `tanh` to enable
    simplification of hyperbolic functions under assumptions.

### CODE REVIEW

File: sympy/assumptions/refine.py [hunk 1]
```diff
@@ -533,6 +533,82 @@ def refine_sin_cos(expr, assumptions):
...
+        if ask(Q.imaginary(arg), assumptions):
+            # sinh(I*y) = I*sin(y) for real y
+            from sympy import I, sin
+            coeff = arg / I
+            if ask(Q.real(coeff), assumptions):
+                return I * sin(coeff)
+    return expr
+
+
+def refine_cosh(expr, assumptions):
```
Comment (JosephMehdiyev) [inline sympy/assumptions/refine.py:571]: this reuses so many code of `refine_sinh`, you should avoid this by either calling it internally or combining both of the methods

File: sympy/assumptions/refine.py [hunk 2]
```diff
@@ -533,6 +533,82 @@ def refine_sin_cos(expr, assumptions):
...
+            # cosh(I*y) = cos(y) for real y
+            from sympy import I
+            from sympy.functions.elementary.trigonometric import cos
+            coeff = arg / I
+            if ask(Q.real(coeff), assumptions):
+                return cos(coeff)
+    return expr
+
+
+def refine_tanh(expr, assumptions):
```
Comment (JosephMehdiyev) [inline sympy/assumptions/refine.py:574]: IF `tanh` is implemented `coth` should be implemented as well in my opinion

File: sympy/assumptions/refine.py [hunk 3]
```diff
@@ -533,6 +533,82 @@ def refine_sin_cos(expr, assumptions):
...
+    if ask(Q.zero(arg), assumptions):
+        return S.Zero
+    if ask(Q.real(arg), assumptions):
+        if ask(Q.imaginary(arg), assumptions):
+            # sinh(I*y) = I*sin(y) for real y
+            from sympy import I, sin
+            coeff = arg / I
+            if ask(Q.real(coeff), assumptions):
+                return I * sin(coeff)
+    return expr
```
Comment (JosephMehdiyev) [inline sympy/assumptions/refine.py:566]: This should handle way more cases, at least very common identities. i.e `sin(-x) =- sin(x)`

Comment (HITESH-S-P) [reply sympy/assumptions/refine.py:566]: working on it.Thanks for the feedback!

Comment (HITESH-S-P) [reply sympy/assumptions/refine.py:566]: 1.Refactored sinh, cosh, tanh, and coth handlers to share a common refine_hyperbolic helper to eliminate code duplication.
2.Added coth support and corresponding tests in test_refine.py.
3.Fixed pure imaginary argument checks and added tests for basic identities and edge cases.

**Comment by HITESH-S-P** [comment]
@JosephMehdiyev I've updated the implementation to address your feedback. All CI checks are passing, could you please take another look when you have a chance?

File: sympy/assumptions/refine.py [hunk 4]
```diff
@@ -533,6 +533,49 @@ def refine_sin_cos(expr, assumptions):
         refined_pow = refine_Pow(pow_expr, assumptions)
         return (pow_expr if refined_pow is None else refined_pow) * sin(rem)
 
+def refine_hyperbolic(expr, assumptions):
```
Comment (JosephMehdiyev) [inline sympy/assumptions/refine.py:536]: as I have told you before, this should handle more cases. You can search hyperbolic identities in a search engine.

File: sympy/assumptions/refine.py [hunk 5]
```diff
@@ -617,4 +658,8 @@ def refine_floor_ceiling(expr, assumptions):
     'Heaviside': refine_Heaviside,
     'floor': refine_floor_ceiling,
     'ceiling' : refine_floor_ceiling,
+    'sinh': refine_sinh,
+    'cosh': refine_cosh,
+    'tanh': refine_tanh,
+    'coth': refine_coth,
```
Comment (JosephMehdiyev) [inline sympy/assumptions/refine.py:664]: what is the point of all these new methods if all the methods just call `refine_hyperbolic`?  I did not mean this at all.
